testsuite/forge/src/backend/k8s/swarm.rs

```diff
@@ -358,47 +510,67 @@ pub fn clean_k8s_cluster(
-            .stdout(Stdio::inherit())
-            .args(&validator_upgrade_args)
-            .output()
-            .expect("failed to helm upgrade diem");
-        assert!(validator_upgrade_output.status.success());
-    });
+    assert!(
+        testnet_upgrade_output.status.success(),
+        "{}",
+        String::from_utf8(testnet_upgrade_output.stderr).unwrap()
+    );
 
-    let kube_client = rt.block_on(K8sClient::try_default()).unwrap();
+    let rt = Runtime::new().unwrap();
 
-    rt.block_on(async {
+    // wait for genesis to run again, and get the updated validators
+    let mut validators = rt.block_on(async {
+        let mut kube_proxy = Command::new(KUBECTL_BIN).arg("proxy").spawn().unwrap();
         diem_retrier::retry_async(k8s_retry_strategy(), || {
-            let jobs: Api<Job> = Api::namespaced(kube_client.clone(), "default");
             Box::pin(async move {
-                let job_name = format!("diem-testnet-genesis-e{}", new_era);
-                println!("Running get job: {}", &job_name);
-                let genesis_job = jobs.get_status(&job_name).await.unwrap();
-                println!("Status: {:?}", genesis_job.status);
-                let status = genesis_job.status.unwrap();
-                match status.succeeded {
-                    Some(1) => {
-                        println!("Genesis job completed");
-                        Ok(())
-                    }
-                    _ => bail!("Genesis job not completed"),
-                }
+                debug!("Running local kube pod healthcheck on {}", HEALTH_CHECK_URL);
+                reqwest::get(HEALTH_CHECK_URL).await?.text().await?;
+                println!("Local kube pod healthcheck passed");
+                Ok::<(), reqwest::Error>(())
             })
         })
         .await
-    })
+        .expect("Failed kube proxy healthcheck");
+        let config = Config::new(
+            reqwest::Url::parse(HEALTH_CHECK_URL).expect("Failed to parse kubernetes endpoint url"),
+        );
+        let kube_client = K8sClient::try_from(config).unwrap();
+        wait_genesis_job(&kube_client, new_era).await.unwrap();
+        let vals = get_validators(kube_client.clone()).await.unwrap();
+        kube_proxy.kill().unwrap();
+        vals
+    });
+
+    // healthcheck on each of the validators
+    let unhealthy_validators = validators
+        .iter_mut()
+        .filter_map(|(_, val)| {
+            let val_name = val.name.clone();
+            println!("Attempting health check: {}", val_name);
+            // perform healthcheck with retry, returning unhealthy
+            let check = diem_retrier::retry(k8s_retry_strategy(), || match val.health_check() {
+                Ok(_) => {
+                    println!("Validator {} healthy", val_name);
+                    Ok(())
+                }
+                Err(ref x) => {
+                    println!("Validator {} unhealthy: {}", val_name, x);
```

Comment:
I think we might do not need this print out which I expect it will be very annoy + we do printout unhealthy val below

Reply:
Validators shouldn't take too long to spin up since we're doing it in parallel on k8s. This is more of a liveness check and makes it easy to compare the logs to actual kube state. I'd prefer if we kept something like this.